ui/src/app/store/extension/extension.service.ts

```diff
@@ -1,13 +1,23 @@
 import { Injectable } from '@angular/core';
 
+import { Http, Response, ResponseContentType } from '@angular/http';
 import { Restangular } from 'ngx-restangular';
 
 import { RESTService } from '../entity/rest.service';
 import { Extension, Extensions } from '../../model';
 
 @Injectable()
 export class ExtensionService extends RESTService<Extension, Extensions> {
-  constructor(restangular: Restangular) {
+  constructor(restangular: Restangular, private http: Http) {
     super(restangular.service('../v1beta1/extensions'), 'extension');
   }
+
+  public getUploadUrl() {
+    return this.restangularService.one().getRestangularUrl();
+  }
+
+  public importExtension(id: string) {
```

Comment:
I miss the type annotation in these methods, so one can spot  at a glance what do they return instead of having to inspect them through the IDE's tschecker or looking into the method body.

Reply:
Sure!